OOP/oop_home_task.py

```diff
@@ -0,0 +1,475 @@
+        """
+        Blank method for publishing of Advertisement that will be overwritten
+        @return: nothing
+        """
+        pass
+
+
+class Article(Publication):
+    """
+    Class that represents the Article publication type. Inherits Publication class.
+    """
+    def __init__(self, publication_text, author, publication_date=None, publication_type=3, city=None):
+        """
+        Method is constructor for Advertisement class
+        @param publication_text: text of publication
+        @param author: author of publication
+        @param publication_date: date of publication
+        @param publication_type: type of publication
+        @param city: city where something happened
+        """
+        self.author = author
+        self.article_splitter = 'Article ----------------------'
+        super().__init__(publication_date, publication_type, city, publication_text)
+
+    @staticmethod
+    def calculate_symbols_count(publication_text):
+        """
+        Method that calculated total of alphanumeric symbols in text entered as Article.
+        @param publication_text: text of publication
+        @return: total symbols count
+        """
+        symbols_count = 0
+        for symbol in publication_text:
+            if symbol.isalnum():
+                symbols_count += 1
+        return symbols_count
+
+    def publish_article(self):
+        """
+        Blank method for publishing of Article that will be overwritten
+        @return: nothing
+        """
+        pass
+
+
+class ObjectsCreator:
+    """
+    Class for objects creation
+    """
+    @staticmethod
+    def create_news(city, publication_date, publication_text):
+        """
+        Creates a new news object with specified set of parameters
+        @param city: city where something happened
+        @param publication_date: date of publication
+        @param publication_text: text of publication
+        @return: created news object
+        """
+        news = News(city, publication_date, publication_text)
+        return news
+
+    @staticmethod
+    def create_advertisement(publication_date, expiration_date, publication_text):
+        """
+        Creates a new advertisement object with specified set of parameters
+        @param publication_date: date of publication
+        @param expiration_date: date when adv will expire
+        @param publication_text: text of publication
+        @return: created advertisement object
+        """
+        advertisement = Advertisement(publication_date, expiration_date, publication_text)
+        return advertisement
+
+    @staticmethod
+    def create_article(publication_text, author):
+        """
+        Creates a new article object with specified set of parameters
+        @param publication_text: text of publication
+        @param author: author of article
+        @return: created article object
+        """
+        article = Article(publication_text, author)
+        return article
```

Comment:
Not sure if we need this class at all, why don't just create new objects in the code without class?

Reply:
Removed.